kie-server-parent/kie-server-services/kie-server-services-openshift/src/main/java/org/kie/server/services/openshift/impl/storage/cloud/KieServerStateOpenShiftRepository.java

```diff
@@ -184,6 +187,41 @@ public KieServerState load(@NotNull String serverId) {
         return kieServerState;
     }
     
+    public String getAppName(String serverId, OpenShiftClient client) {
+        DeploymentConfig dc = getKieServerDC(serverId, client);
+        return dc != null ? dc.getMetadata().getLabels().get(CFG_MAP_LABEL_APP_NAME) : getAppNameByHostname();
+    }
+    
+    public String getAppNameByHostname() {
+        String hostname = System.getenv(ENV_HOSTNAME);
+        return hostname != null ? hostname.split("-")[0] : null;
```

Comment:
This is a bit risky as it won't work in case application name contains '-'

Reply:
@sutaakar @ruromero 
Agree. It is not ideal. We have identified a better solution. Since HOSTNAME is always equivalent to Pod name, it can be used as the lookup key for retrieving the hosting BC/WB Pod, and in turn, to get the RHPAM application name from Pod's label. (See below) 

```
$ oc get pod myapp-rhpamcentr-2-mnp7g -L application
NAME                       READY     STATUS    RESTARTS   AGE       APPLICATION
myapp-rhpamcentr-2-mnp7g   1/1       Running   0          1h        myapp
$ oc exec myapp-rhpamcentr-2-mnp7g -- hostname
myapp-rhpamcentr-2-mnp7g
```